lib/jormungandr/src/Cardano/Wallet/Jormungandr/Api.hs

```diff
@@ -146,3 +189,27 @@ data Hex
 -- | Represents data rendered to hexadecimal text.
 instance Accept Hex where
     contentType _ = contentType $ Proxy @Servant.PlainText
+
+{-------------------------------------------------------------------------------
+                            FromJson instances
+-------------------------------------------------------------------------------}
+
+instance FromJSON StakeApiResponse where
+    parseJSON = genericParseJSON defaultOptions
+
+instance FromJSON (ApiStakeDistribution) where
+    parseJSON = genericParseJSON defaultOptions
+
+instance FromJSON (ApiT Stake) where
+    parseJSON (Number n) = case toBoundedInteger n of
+        Nothing ->
+            fail "stake should be non-negative integer"
+        Just s ->
+            return (ApiT $ Stake $ Quantity s)
+    parseJSON _ = fail "stake should be numeric"
```

Comment:
What about:

```hs
parseJSON = fmap ApiT . genericParseJSON defaultRecordTypeOptions
```

?

Reply:
I will get something like : `expected unary record, encountered Number`

Reply:
But we do expect a record though, we represent amounts as `{ "quantity": ..., "unit": ... }` , right ?

Reply:
well we need to parse stuff like 
```
$ curl http://127.0.0.1:8080/api/v0/stake
{"epoch":252054,"stake":{"dangling":0,"pools":[["7d749ef424507fb80fed0d2289d535a94f6870add0cf8b374cfe6cae078320ec",1]],"unassigned":100100000000000}}
```
Of course I can do it using : 
```
204 instance FromJSON (ApiT (Quantity "lovelace" Word64)) where                                                                                                                               
205     parseJSON = fmap (ApiT . Quantity) . parseJSON
```
I loose error msgs that resonate with me, but of course do not need to import `scientific`. Eventually went for the solution ^^^